### ISSUE TITLE
Array-style validation rules extensions

### ISSUE THREAD

**Issue by zlodes** [laravel/ideas#2453]
Now when I want to create my own class-based valuidation rule I need to write something like that:

```php
Validator::extend('my-awesome-rule', MyRuleClass::class . '@validate');
``` 

This is very similar to rotues declaration:
```php
Route::post('/something', [SomethingController::class, 'doSomething']);
```

I think that will be cool to have the same syntax for such things. 

What you think about it?

P.S.: I can do PR.

**Comment by akbarjimi** [comment]
i agree with you

**Comment by jeroenvanrensen** [comment]
Do you mean you can use this code?

```php
Validator::extend('my-awesome-rule', [MyRuleClass::class, 'validate']);
```

**Comment by zlodes** [comment]
@jeroenvanrensen Yes. I want, but now it isn't possible.

**Comment by jeroenvanrensen** [comment]
@zlodes But why don't you just use this?

```php
request()->validate([
    'name' => ['required', MyRule::class]
]);
```

**Comment by aarreedd** [comment]
@jeroenvanrensen because that is not as elegant and @zlodes wants to use `extend()`.

Why not declare all rules using the class name? Because using the string identifier (e.g. "required" or "my-awesome-rule") is nicer.

**Comment by jeroenvanrensen** [comment]
Yeah I agree but I personally won't use it because it is less clear what the default and custom rules are.

**Comment by zlodes** [comment]
@jeroenvanrensen because it has different definition. 
### Class-based rules:
https://laravel.com/docs/8.x/validation#using-rule-objects
It can be useful for simple rules without params.

### Closure-based rules
https://laravel.com/docs/8.x/validation#using-closures
Seems bad, hard to test. Also doesn't have params and validator instance.

### Method-based rules
```php

class MyRuleClass 
{
    public function validate(string $attribute, mixed $value, array $parameters, Validator $validator): bool
    {
        // Here you can access to rule parameters and validator instance (to get validating data, for example).
        return false;
    }
}

// ...

Validator::extend(
    'my-awesome-rule', 
    MyRuleClass::class . '@validate', 
    'Value for field :attribute is so bad...'
);
```

And it works fine, but the extending with `MyRuleClass::class . '@validate'` looks not so good.

**Comment by jeroenvanrensen** [comment]
@zlodes But what is the difference bewteen class based rules and method based rules? As far as I can see they have the same functionality, right?

**Comment by zlodes** [comment]
@jeroenvanrensen not the same. The difference in the validation method params. 

Class-based rule that implementing `Rule` interface has just only attribute name and its value.

**Comment by ryangjchandler** [comment]
> @jeroenvanrensen not the same. The difference in the validation method params.
> 
> Class-based rule that implementing `Rule` interface has just only attribute name and its value.

Arguments can be passed in using the constructor, e.g.

```php
request()->validate([
    'name' => ['required', new MyRule('argument one', true)]
]);
```

**Comment by zlodes** [comment]
@ryangjchandler yes. 

But a class-based rule has no access to the validator instance and the data under validation (e.g. for validating the field against another field).
